packages/twenty-server/src/engine/core-modules/code-engine/utils/create-zip-file.ts

```diff
@@ -0,0 +1,18 @@
+import fs from 'fs';
+
+import archiver from 'archiver';
+
+export const createZipFile = async (sourceDir, outPath): Promise<void> => {
+  const output = fs.createWriteStream(outPath);
+  const archive = archiver('zip', {
+    zlib: { level: 9 }, // Compression level
+  });
+
+  return new Promise((resolve, reject) => {
+    output.on('close', () => resolve());
+    archive.on('error', (err) => reject(err));
+    archive.pipe(output);
+    archive.directory(sourceDir, false);
+    archive.finalize();
+  });
```

Comment:
When dealing with streams pipelines, I like to use the [pipeline](https://nodejs.org/api/stream.html#streampipelinestreams-options) function from the `node:stream` module, which automatically forwards errors and returns a Promise that is resolved when the whole pipeline is finished.

I didn't tested the following code block but that's what I'm thinking about:

```suggestion
  const p = pipeline(archive, output)

  archive.directory(sourceDir, false);
  archive.finalize();

  return p;
```

I looove playing with streams when developing with Node.js, and I love seeing your code 😍

Reply:
ok I will try that